src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -295,6 +328,94 @@ void ScheduleTask(DateTimeOffset utcNow)
             }
         }
 
+        private (long, long) Sum()
+        {
+            lock (_allStats!)
+            {
+                long hits = _accumulatedStats!.Hits;
+                long misses = _accumulatedStats.Misses;
+
+                foreach (WeakReference<Stats> wr in _allStats)
+                {
+                    if (wr.TryGetTarget(out Stats? stats))
+                    {
+                        hits += Interlocked.Read(ref stats.Hits);
+                        misses += Interlocked.Read(ref stats.Misses);
+                    }
+                }
+
+                return (hits, misses);
+            }
+        }
+
+        private void Hit()
+        {
+            if (_allStats is not null)
+            {
+                if (IntPtr.Size == 4)
+                    Interlocked.Increment(ref GetStats().Hits);
+                else
+                    GetStats().Hits++;
+            }
+        }
+
+        private void Miss()
+        {
+            if (_allStats is not null)
+            {
+                if (IntPtr.Size == 4)
+                    Interlocked.Increment(ref GetStats().Misses);
+                else
+                    GetStats().Misses++;
+            }
+        }
+
+        private Stats GetStats() => _stats!.Value!;
+
+        internal sealed class Stats
+        {
+            public long Hits;
+            public long Misses;
+            private readonly MemoryCache? _memoryCache;
+            public Stats() { }
+            public Stats(MemoryCache memoryCache)
+            {
+                _memoryCache = memoryCache;
+                _memoryCache.AddToStats(this);
+            }
+            ~Stats()
+            {
+                _memoryCache?.RemoveFromStats(this);
+            }
```

Comment:
Nit: some blank lines separating things would be nice:
```suggestion
            private readonly MemoryCache? _memoryCache;
            public long Hits;
            public long Misses;

            public Stats() { }

            public Stats(MemoryCache memoryCache)
            {
                _memoryCache = memoryCache;
                _memoryCache.AddToStats(this);
            }

            ~Stats() => _memoryCache?.RemoveFromStats(this);
```